apso/src/main/scala/eu/shiftforward/apso/config/Implicits.scala

```diff
@@ -276,23 +276,28 @@ object Implicits extends BasicConfigReaders with ExtendedConfigReaders {
      * @return the Map value
      */
     def toMap[T](implicit configReader: ConfigReader[T]): Map[String, T] = {
-      // config key strings are different between `conf.root.entrySet` and `conf.entrySet()`
-      val unwrappedSet = conf.root.entrySet().map(e => e.getKey -> e.getValue)
-      val wrappedSet = conf.entrySet().map(e => e.getKey -> e.getValue)
-
-      unwrappedSet.flatMap {
-        case (k, v) =>
-          v match {
-            case obj: ConfigObject => obj.toConfig().toMap(configReader).map {
-              case (nk, nv) =>
-                (k + "." + nk) -> nv
+
+      def aux(conf: Config, configReader: ConfigReader[T]): Map[Seq[String], T] = {
```

Comment:
You don't need to pass `configReader` around. It's always going to be the same under the `toMap` method.